packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -255,45 +258,41 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
                         headers: Object.keys(headers).length !== 0 ? true : undefined,
                         waitQueueLength: this.epochTracker.rateLimiter.waitQueueLength,
                     },
-                    async () => this.epochTracker.fetchAndParseAsJSON<IBlob>(url, { headers }, FetchType.blob),
+                    async () => this.epochTracker.fetchResponse(url, { headers }, FetchType.blob),
                 );
             });
-            blob = response.content;
+            buffer = await response.arrayBuffer();
         }
 
-        assert(blob.encoding === "base64" || blob.encoding === undefined, "wrong blob encoding format");
-        const inputFormat = blob.encoding;
-        if (!this.attributesBlobHandles.has(blobid)) {
-            if (outputFormat === inputFormat || (outputFormat === "string" && inputFormat === undefined))  {
-                return blob.content;
-            }
-            else if (outputFormat === "base64") {
-                return fromUtf8ToBase64(blob.content);
-            }
-            else {
-                return fromBase64ToUtf8(blob.content);
+        if (!this.attributesBlobHandles.has(blobId)) {
+            if (blob) {
+                if (outputFormat === blob.encoding || (outputFormat === "string" && blob.encoding === undefined))  {
+                    return blob.content;
+                } else if (outputFormat === "base64") {
+                    return fromUtf8ToBase64(blob.content);
+                } else {
+                    return fromBase64ToUtf8(blob.content);
+                }
             }
+
+            return IsoBuffer.from(buffer).toString(outputFormat === "base64" ? "base64" : "utf8");
```

Comment:
This does not look right. transformation of attributes content should happen in the same way, no matter if it blob came from cache or not.

Reply:
Fixing the branch name is done below, not here (unless I'm misunderstanding you)